diff --git a/src/dwarf/Gfind_proc_info-lsb.c b/src/dwarf/Gfind_proc_info-lsb.c
index 0cfb4d4f..644c5560 100644
--- a/src/dwarf/Gfind_proc_info-lsb.c
+++ b/src/dwarf/Gfind_proc_info-lsb.c
@@ -127,6 +127,7 @@ load_debug_frame (const char *file, char **buf, size_t *bufsize, int is_local)
       return 1;
     }
 
+#if defined(SHF_COMPRESSED)
   if (shdr->sh_flags & SHF_COMPRESSED)
     {
       unsigned long destSize;
@@ -160,6 +161,7 @@ load_debug_frame (const char *file, char **buf, size_t *bufsize, int is_local)
         }
     }
   else
+#endif
     {
       *bufsize = shdr->sh_size;
       GET_MEMORY(*buf, *bufsize);
